facenet_sandberg/generate_pairs.py

```diff
@@ -4,139 +4,124 @@
-
-
-def write_pairs(fname: str,
-                match_sets: List[List[Tuple[str,
-                                            int,
-                                            int]]],
-                mismatch_sets: List[List[Tuple[str,
-                                               int,
-                                               str,
-                                               int]]],
-                k_num_sets: int,
-                total_matches_mismatches: int) -> None:
-    file_contents = f'{k_num_sets}\t{total_matches_mismatches}\n'
-    for match_set, mismatch_set in zip(match_sets, mismatch_sets):
-        for match in match_set:
-            file_contents += f'{match[0]}\t{match[1]}\t{match[2]}\n'
-        for mismatch in mismatch_set:
-            file_contents += f'{mismatch[0]}\t{mismatch[1]}\t{mismatch[2]}\t{mismatch[3]}\n'
-
-    with open(fname, 'w') as fpairs:
-
-        fpairs.write(file_contents)
+            if person1_images and person2_images:
+                img1 = int(''.join([i for i in random.choice(person1_images)
+                                    if i.isnumeric()]).lstrip('0'))
+                img2 = int(''.join([i for i in random.choice(person2_images)
+                                    if i.isnumeric()]).lstrip('0'))
+                if person1.lower() > person2.lower():
+                    person1, img1, person2, img2 = person2, img2, person1, img1
+                mismatch = (person1, img1, person2, img2)
+                if mismatch not in mismatches:
+                    mismatches.append(mismatch)
+                    curr_matches += 1
+    return sorted(mismatches, key=lambda x: x[0].lower())
+
+
+def _parse_arguments(argv):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--image_dir',
+                        type=str,
+                        required=True,
+                        help='Path to the image directory.')
+    parser.add_argument('--pairs_file_name',
+                        type=str,
+                        required=True,
+                        help='Filename of pairs.txt')
+    parser.add_argument('--num_folds',
+                        type=int,
+                        required=True,
+                        help='Number of folds for k-fold cross validation.')
+    parser.add_argument('--num_matches_mismatches',
+                        type=int,
+                        required=True,
+                        help='Number of matches/mismatches per fold.')
+    return parser.parse_args(sys.argv[1:])
 
 
 if __name__ == '__main__':
-    # image_dir = os.path.join(
-    total_matches_mismatches = 15
-    # image_dir = os.path.join(
-    #                os.path.dirname(
-    #                    os.path.abspath(__file__)
-    #                ),
-    #            'images')
-    image_dir = '/home/miperel/redcross/facenet/datasets/lfw/raw_mtcnn'
-
-    people_lists = split_people_into_sets(image_dir, k_num_sets)
+    args = _parse_arguments(sys.argv[1:])
```

Comment:
Nit: I've seen it more often that you have a `_cli` function that gets called from `__main__` which does the parsing, unpacks the args and then delegates to a `_main` function which performs the actual logic. This enables a user to drive the CLI functionality from other scripts (e.g. that's how Celery works) and also prevents the variables assigned in `__main__` from leaking into the global scope.